runtime/runtime/src/config.rs

```diff
@@ -120,6 +120,7 @@ pub fn total_send_fees(
             },
             DeleteKey(_) => cfg.delete_key_cost.send_fee(sender_is_receiver),
             DeleteAccount(_) => cfg.delete_account_cost.send_fee(sender_is_receiver),
+            Delegate(_) => cfg.delegate_cost.send_fee(sender_is_receiver),
```

Comment:
I think it's about time we start talking about gas fees.

There are 3 places where `validate_actions` loops through all inner actions and validates them.
1. When validating the transaction on the relayer shard.
2. When validating the receipt with the delegate action at Alice's shard.
3. When validating the final receipt at FT's shard.

The `send` fee of an action is meant to cover the cost of validation at the sender side and the conversion to a receipt. The `exec` fee covers validation and execution of the action on the receiver. But here we have a step in between, which makes things a bit more complicated. One approach would be to pay the send fee for inner actions twice, in step 1 & 2 but this raises the cost of an FT by 2.3 Tgas, almost 50%. Not sure if we have a better option though.

If I got this right, the current implementation does not include the send cost of inner actions at all. When the `tx_cost` is computed in step 1, the inner actions' send fees are not added, only the constant send fee for delegate action itself. And in step 2, when the receipt is created, the prepaid exec fees for inner actions is calculated and added to used gas, which is correct, but the send fees are not calculated and burnt as they should be.

Do I make sense? And have you already thought about this?

Reply:
>Do I make sense? And have you already thought about this?

I thought about it and I've just realized that I misunderstood the purposes of `send` and `exec` fees.
I didn't include `send` fee because I thought the inner actions' fee should be included _only_ to `exec` fee (they are converted on the receiver side therefore it should be include to `exec` fee). That was my idea.

Reply:
Yeah it's easy to think like you did. It's also not documented all that well. But fact is, for normal transaction we charge `send` and `exec` fees for all actions included and at the very least we have to do the same for delegated actions.

I would suggest to charge the send fee twice for delegated actions. But if you have other suggestions I am happy to discuss them.